benchmark/fs/bench-mkdtempSync.js

```diff
@@ -0,0 +1,39 @@
+'use strict';
+
+const common = require('../common');
+const fs = require('fs');
+const tmpdir = require('../../test/common/tmpdir');
+
+const bench = common.createBenchmark(main, {
+  type: ['valid-string', 'valid-buffer', 'invalid'],
+  n: [1e3],
+});
+
+function main({ n, type }) {
+  tmpdir.refresh();
+  let prefix;
+
+  switch (type) {
+    case 'valid-string':
+      prefix = tmpdir.resolve(`${Date.now()}`);
+      break;
+    case 'valid-buffer':
+      prefix = Buffer.from(tmpdir.resolve(`${Date.now()}`));
+      break;
+    case 'invalid':
+      prefix = tmpdir.resolve('non-existent', 'foo', 'bar');
+      break;
+    default:
+      new Error('Invalid type');
+  }
+
+  bench.start();
+  for (let i = 0; i < n; i++) {
+    try {
+      fs.mkdtempSync(prefix, { encoding: 'utf8' });
```

Comment:
Store the result and just `assert.ok` after the benchmark, just to avoid deadcode elimination.

Reply:
Also, I think you can cache the options.

Reply:
Definitely! Thank you for leaving a review. Updating the PR.